File: server/commits.py
import requests
from datetime import datetime, date, timedelta
    
from dotenv import dotenv_values

def has_commited():
    env = dotenv_values(".env")  # config = {"USER": "foo", "EMAIL": "foo@example.org"}
    repo_api_url = "https://api.github.com/user/repos?per_page=100"
    date_string = str((datetime.today() - timedelta(days=1))).split(" ")[0] + "T21:00:00.000000"
    bearer_token = env['GITHUB_OAUTH']
    headers={"Authorization":bearer_token}
    repo_json = requests.get(repo_api_url, headers=headers)
    repos = repo_json.json()
    for repo in repos:
        commit_api_url = "https://api.github.com/repos/zrwaite/" + str(repo['name']) + "/commits"
        commit_api_params = {"author":"zrwaite","since": date_string}
        commit_json = requests.get(commit_api_url, headers=headers, params=commit_api_params)
        commits = commit_json.json()
        if len(commits) != 0:
            return True
    return False

print(has_commited())

File: src/commits.ts
import axios from "axios";
import env from "dotenv";
env.config();

const bearerToken = process.env.GITHUB_OAUTH||"";
const headers={"Authorization":bearerToken}

const hasCommitted = async () => {
	const repoApiUrl = "https://api.github.com/user/repos?per_page=100";
	const date = new Date();
	date.setHours(date.getHours() - 5);
	const reposResponse = await axios.get(repoApiUrl, {headers:headers});
	const repos = reposResponse.data; 
	const dateString = date.toISOString().split("T")[0]+"T05:00:00.000Z";
	console.log(dateString);
	for (let i=0; i<repos.length; i++) {
		const repo = repos[i];
        const commitApiUrl = "https://api.github.com/repos/zrwaite/" + repo.name + "/commits"
        const commitApiParams = {"author":"zrwaite","since": dateString}
		const commitResponse = await axios.get(commitApiUrl, {headers:headers, params:commitApiParams})
		const commits = commitResponse.data;
		if (commits.length !== 0) return true;
	}
	return false;
}


export {hasCommitted}